Drop support for RSA key generation

umayr opened the pull request:
After this change notary won't be able to generate RSA keys, although it can import and parse keys if provided externally. 

CC: @endophage

dingwilson commented:
Is there any timeline for when RSA support will be dropped?

endophage commented:
@dingwilson we have no intent to drop support for using externally provided RSA keys. We are only ending support for internally generating RSA keys.

riyazdf commented on client/client_test.go:
```diff
@@ -194,7 +195,7 @@ func createRepoAndKey(t *testing.T, rootType, tempBaseDir, gun, url string) (*No
 	require.NoError(t, err, "error creating repo: %s", err)
 
-	rootPubKey, err := repo.CryptoService.Create(data.CanonicalRootRole, repo.gun, rootType)
+	rootPubKey, err := testutils.CreateOrAddKey(repo.CryptoService, data.CanonicalRootRole, repo.gun, rootType)
```
ah of course – because the cryptoservice cannot create RSA keys?  If that is the exact reason, this is perfectly fine then.

umayr replied:
@riyazdf Exactly. 

To add more on this, I tried to add keys to crypto service in case of RSA, while for other sort of keys I created them with crypto service which later adds them internally.

umayr commented:
@riyazdf Rebased and fixed the conflicts. Let me know if there's anything else.

umayr commented:
@endophage I have added an explicit error message that tells rsa key generation is not supported, I have also added a negative case that verifies this case moreover, I have rebased it with latest origin/master. Please let me know it requires anything else.
